src/app/brukerprofil/kontaktinformasjon/TilrettelagtKommunikasjonForm.tsx

```diff
@@ -0,0 +1,161 @@
+import * as React from 'react';
+import { FormEvent } from 'react';
+import { Action } from 'history';
+import { connect, Dispatch } from 'react-redux';
+import styled from 'styled-components';
+
+import Undertittel from 'nav-frontend-typografi/lib/undertittel';
+import KnappBase from 'nav-frontend-knapper';
+
+import { STATUS } from '../../../redux/utils';
+import { AppState } from '../../../redux/reducer';
+import { VeilederRoller } from '../../../models/veilederRoller';
+import {
+    EndreTilrettelagtKommunikasjonrequest
+} from
+        '../../../redux/brukerprofil/endreTilrettelagtKommunikasjonrequest';
+import { endreTilrettelagtKommunikasjon, reset } from '../../../redux/brukerprofil/endreTilrettelagtKommunikasjon';
+import { Person } from '../../../models/person/person';
+import CheckboksPanelGruppe from 'nav-frontend-skjema/lib/checkboks-panel-gruppe';
+import { CheckboksProps } from 'nav-frontend-skjema/src/checkboks-panel';
+import { KodeverkResponse } from '../../../models/kodeverk';
+import RequestTilbakemelding from './RequestTilbakemelding';
+
+const SubmitknappWrapper = styled.div`
+  margin-top: 1em;
+`;
+
+interface State {
+    checkbokser: CheckboksProps[];
+}
+
+interface DispatchProps {
+    endreTilrettelagtKommunikasjon: (request: EndreTilrettelagtKommunikasjonrequest) => void;
+    resetEndreTilrettelagtKommunikasjonReducer: () => void;
+}
+
+interface StateProps {
+    status: STATUS;
+}
+
+interface OwnProps {
+    person: Person;
+    veilederRoller?: VeilederRoller;
+    tilrettelagtKommunikasjonKodeverk: KodeverkResponse;
+}
+
+type Props = DispatchProps & StateProps & OwnProps;
+
+class TilrettelagtKommunikasjonsForm extends React.Component<Props, State> {
+
+    constructor(props: Props) {
+        super(props);
+        this.state = {
+            checkbokser: this.lagKnapper()
+        };
+        this.handleSubmit = this.handleSubmit.bind(this);
+        this.handleOnChange = this.handleOnChange.bind(this);
+    }
+
+    lagKnapper() {
+        const tilrettelagtKommunikasjonKodeverk = this.props.tilrettelagtKommunikasjonKodeverk.kodeverk;
+        return tilrettelagtKommunikasjonKodeverk.map((kommunikasjonsmetode) => {
+            return {
+                label: kommunikasjonsmetode.beskrivelse,
+                value: kommunikasjonsmetode.value,
+                id: kommunikasjonsmetode.kodeRef,
+                checked: this.props.person.tilrettelagtKomunikasjonsListe.some((tk) =>
+                    tk.behovKode === kommunikasjonsmetode.kodeRef),
+                disabled: !this.harVeilderPåkrevdRolle()
+            };
+        });
+    }
+
+    handleSubmit(event: FormEvent<HTMLFormElement>) {
+        this.props.endreTilrettelagtKommunikasjon({
+            fødselsnummer: this.props.person.fødselsnummer // TODO må sende inn en gyldig request her
```

Comment:
her er en TODO som ikke er fikset/fjernet.

Reply:
og er egentlig dette riktig rolle for å endre tilrettelagt kommunikasjon?

Reply:
jepp, TODO'en venter på at backenden er ferdig før vi vet hvordan denne requesten skal lages.

aner ikke om det er riktig rolle nei, det må vi ta en gjennomgang å sjekke til slutt tenker jeg :-)

Reply:
Jeg er heller fan av å lage issues enn å legge inn TODO i koden btw 👑 🐼